test/e2e-common.sh

```diff
@@ -0,0 +1,31 @@
+#!/usr/bin/env bash
+
+# Copyright 2019 The Knative Authors
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+function cluster_setup() {
+  header "Building client"
+  ${REPO_ROOT_DIR}/hack/build.sh -f || return 1
+}
+
+function knative_setup() {
```

Comment:
Is the invocation of this controlled by the flag?

Reply:
This function is invoked from ./vendor/knative.dev/test-infra/scripts/e2e-tests.sh . This script is imported in e2e-tests.sh and tekton-tests.sh so it's called in both scripts. I'm not sure which flag you mean.

Reply:
I was hoping to make sure that there's a way to NOT have the script try to install Knative and just assume that the person running the e2e2 tests already installed all of the prereqs into the cluster.  In our testing env we need to control which bits are installed/tested - we can't let the script choose things for us.

Reply:
Alright, I think you can still control that by passing `--skip-knative-setup` . See https://github.com/knative/client/blob/master/vendor/knative.dev/test-infra/scripts/e2e-tests.sh#L376 and https://github.com/knative/client/blob/master/vendor/knative.dev/test-infra/scripts/e2e-tests.sh#L459 .

Reply:
That's correct. Running e2e-tests.sh with --skip-knative-setup will make this function a no-op.